util/concurrenz/wait_group.go

```diff
@@ -0,0 +1,82 @@
+package concurrenz
```

Comment:
I've seen libraries contain functionality like this (i.e. https://caolan.github.io/async/v3/ ). Seems like go will have an explosion of stuff like this with generics.

Reply:
Yeah, from what I've read they're not going do much with the standard libraries in 1.18. Instead they're going to take some more time to make sure they don't rush the library changes. Hopefully stuff like this will eventually be in there, though. I'm guessing some third party libraries will become de facto standards and will then get pulled into core.